### PR TITLE
refactor update `blas/ext/base/dsnansumors` follow current project conventions

### LINKED ISSUE #1493: [RFC]: refactor `blas/ext/base/dsnansumors` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dsnansumors`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dsnansumors) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by rajutkarsh07**
Resolves #1493  .

## Description

> What is the purpose of this pull request?
 update `blas/ext/base/dsnansumors` follow current project conventions

This pull request:

-  progresses #1152 
-  fixes #1493 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by rajutkarsh07** [comment]
@Planeshifter @kgryte @Pranavchiku Please review this PR

**Comment by rajutkarsh07** [comment]
@Planeshifter I have made the required changes please review this

File: lib/node_modules/@stdlib/blas/ext/base/dsnansumors/README.md [hunk 1]
```diff
@@ -94,26 +89,23 @@ Computes the sum of single-precision floating-point strided array elements, igno
 var Float32Array = require( '@stdlib/array/float32' );
 
 var x = new Float32Array( [ 1.0, -2.0, NaN, 2.0 ] );
-var N = x.length;
 
-var v = dsnansumors.ndarray( N, x, 1, 0 );
-// returns 1.0
+var v = dsnansumors.ndarray( 2, x, 1, 0 );
+// returns -1
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansumors/README.md]:
```suggestion
var v = dsnansumors.ndarray( 4, x, 1, 0 );
// returns 1.0
```

**Review by Planeshifter — APPROVED** [review]
Thanks! Had to make a bunch of changes to get CI to pass and C benchmarks / examples to compile, but the PR should now be ready to land.

**Pull request merged**